keep/api/core/db.py

```diff
@@ -2044,6 +2043,42 @@ def get_incident_alerts_by_incident_id(tenant_id: str, incident_id: str) -> List
     return query.all()
 
 
+def get_alerts_data_for_incident(
+    alert_ids: list[str | UUID],
+    session: Optional[Session]
+) -> dict:
+
+    def inner(db_session: Session):
+        alerts_data = db_session.exec(
+            select(
+                func.json_extract(Alert.event, '$.service').label("service"),
+                func.json_extract(Alert.event, '$.source').label("source")
+            ).where(
+                col(Alert.id).in_(alert_ids),
+            )
+        ).all()
```

Comment:
This is a bit problematic because of different sql dialects (looks for `session.bind.dialect.name` across the code and you'll see)